paddle/fluid/framework/details/multi_devices_graph_pass.cc

```diff
@@ -436,13 +456,14 @@ std::unique_ptr<ir::Graph> MultiDevSSAGraphBuilder::ApplyImpl(
       }
     }
   }
-
   bool use_gpu = false;
 #ifdef PADDLE_WITH_CUDA
   use_gpu = nccl_ctxs_ != nullptr;
 #endif
 
-  if (use_gpu && strategy_.reduce_ == BuildStrategy::ReduceStrategy::kReduce) {
+  if ((use_gpu &&
+       strategy_.reduce_ == BuildStrategy::ReduceStrategy::kReduce) ||
+      is_dist_train) {
```

Comment:
Does allreduce need broadcast as well?

Reply:
May not, all the devices have the same date of a variable after allreduce.

Reply:
And no matter reduce or allreduce, we don't need to bcast when distribute training, just sending them to ps, will fix it...